src/ucm/event/event.c

```diff
@@ -286,6 +286,12 @@ void *ucm_shmat(int shmid, const void *shmaddr, int shmflg)
             attach_addr -= attach_addr % SHMLBA;
         }
         ucm_dispatch_vm_munmap((void*)attach_addr, size);
+        ucs_recursive_spin_lock(&ucm_kh_lock);
+        iter = kh_get(ucm_ptr_size, &ucm_shmat_ptrs, (void*)attach_addr);
+        if (iter != kh_end(&ucm_shmat_ptrs)) {
+            kh_del(ucm_ptr_size, &ucm_shmat_ptrs, iter);
+        }
+        ucs_recursive_spin_unlock(&ucm_kh_lock);
```

Comment:
can we unite common code with ucm_shmdt()?

Reply:
united